gdal/swig/python/scripts/gdal2tiles.py

```diff
@@ -462,65 +467,908 @@ def tilefilename(self, x, y, z):
                             "%s-%s-%s.%s" % (z, x, y, self.tileformat))
 
 
-class Gdal2TilesError(Exception):
+class GDALError(Exception):
     pass
 
 
-class GDAL2Tiles(object):
+def exit_with_error(message, details=""):
+    # Message printing and exit code kept from the way it worked using the OptionParser (in case
+    # someone parses the error output)
+    sys.stderr.write("Usage: gdal2tiles.py [options] input_file [output]\n\n")
+    sys.stderr.write("gdal2tiles.py: error: %s\n" % message)
+    if details:
+        sys.stderr.write("\n\n%s\n" % details)
+
+    sys.exit(2)
```

Comment:
I see your point. This is however the existing behavior, albeit hidden behind calls to ArgumentParser.error which does just that.

I'll have a quick look to see how hard it would be to control those "exit conditions" in a different manner.